app/controllers/internal/configs_controller.rb

```diff
@@ -31,6 +32,11 @@ def config_params
     params.require(:site_config).permit(allowed_params)
   end
 
+  def confirmation
+    not_authorized unless current_user.has_role?(:single_resource_admin, Config) # Special additional permission
+    not_authorized if params[:confirmation] != "My username is @#{current_user.username} and this action is 100% safe and appropriate."
```

Comment:
will `not_authorized` on this line tell the user that their confirmation was incorrect? I'm specifically thinking if someone typos this confirmation; we might want to explicitly tell them that their confirmation was not correct (instead of the whatever `not_authorized` responds with?)

Reply:
This is definitely the ideal behavior.

I felt like because of the fairly custom logic going on here with `SiteConfig.public_send` and how possibly destructive this is, I felt kind of nervous about trying to get the UX _just right_ in this step as opposed to just blowing things up at this step if it's not done correctly.

I think a follow-on PR to get this flow proper with plenty of testing to ensure this absolutely can't go wrong would be a good call.